src/client_lib/pegasus_client_impl.h

```diff
@@ -247,13 +247,21 @@ class pegasus_client_impl : public pegasus_client
 
     static void init_error();
 
+    enum class async_scan_type : char
+    {
+        NORMAL,
+        COUNT_ONLY,
+        COUNT_ONLY_FINISHED
+    };
+
     class pegasus_scanner_impl : public pegasus_scanner
     {
     public:
         int next(std::string &hashkey,
                  std::string &sortkey,
                  std::string &value,
-                 internal_info *info = nullptr) override;
+                 internal_info *info = nullptr,
+                 int32_t *count = nullptr) override;
```

Comment:
hashkey-sortkey-value and count will not be returned at the same time? If not, separete it into two functions would be better. one is:
```
        int next(std::string &hashkey,
                 std::string &sortkey,
                 std::string &value,
                 internal_info *info = nullptr)
```
the other is 
```
        int next(int32_t &count,
                 internal_info *info = nullptr)
```